### ISSUE TITLE
BlockExample crash

### ISSUE THREAD

**Issue by cissieAB** [JeffersonLab/JANA2#207]
Built jana as Mar-8-2023 on gluon200 machine, BlockExample crashes because of `BlockExampleSource::DisentangleBlock()`.

```
[hdtrdops@gluon200 ~/GemTrd_2023]$ ./JANA2/install/plugins/BlockExample.so
...
The lines below might hint at the cause of the crash.
You may get help by asking at the ROOT forum https://root.cern/forum
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#6  0x0000000000411ca7 in JFactoryT<MyObject>* JEvent::Insert<MyObject>(MyObject*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const ()
#7  0x000000000040ff96 in BlockExampleSource::DisentangleBlock(MyBlock&, JEventPool&) ()
#8  0x00000000004215b7 in JBlockDisentanglerArrow<MyBlock>::execute(JArrowMetrics&, unsigned long) ()
#9  0x000000000045f960 in JWorker::loop() ()
#10 0x0000000000461b6a in void std::__invoke_impl<void, void (JWorker::*)(), JWorker*>(std::__invoke_memfun_deref, void (JWorker::*&&)(), JWorker*&&) ()
#11 0x0000000000461aac in std::__invoke_result<void (JWorker::*)(), JWorker*>::type std::__invoke<void (JWorker::*)(), JWorker*>(void (JWorker::*&&)(), JWorker*&&) ()
#12 0x0000000000461a1b in void std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) ()
#13 0x00000000004619d6 in std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> >::operator()() ()
#14 0x00000000004619ba in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (JWorker::*)(), JWorker*> > >::_M_run() ()
#15 0x00007f33fa1726c0 in execute_native_thread_routine () at ../../../../../srcdir/libstdc++-v3/src/c++11/thread.cc:80
#16 0x00007f33f9975dd5 in start_thread () from /lib64/libpthread.so.0
#17 0x00007f33f969f02d in clone () from /lib64/libc.so.6


```

**Comment by cissieAB** [comment]
@nathanwbrei  I see you reopen my pull request on the BlockExample CI test, but its still did not pass with the most recent updates. #__#

https://github.com/JeffersonLab/JANA2/actions/runs/4429840220/jobs/7770801856

**Comment by cissieAB** [comment]
I rerun ubuntu22.04 job and it passed: https://github.com/JeffersonLab/JANA2/actions/runs/4429840220/jobs/7772275232

The processed event number is 27. What is the expected processed event number?

**Comment by nathanwbrei** [comment]
27 is correct actually, I think the loop index started at 1 instead of 0

**Comment by nathanwbrei** [comment]
I think I understand the cause of the crash. Situation isn't too bad, I'll push a fix and then we can merge this

**Comment by nathanwbrei** [comment]
This should be fixed in v2.1.0.
